docs/README-develop-best-practices.md

```diff
@@ -55,16 +55,16 @@ All attribute entries in a given YAML file must be kept in alphabetical order. A
 
 This rule is enforced by the `Cisco::CommandReference` class itself - it will raise an exception if it detects any out-of-order entries.
 
-### <a name="yaml3">Y3: Use *regexp* anchors where needed for `config_get` and `config_get_token` entries.
+### <a name="yaml3">Y3: Use *regexp* anchors where needed for CLI `get_context` and `get_value` entries.
 
-Please use *regexp* anchors `^$` to ensure you match the correct feature information in the `show` output.
+By default, CLI clients assume that `get_context` and `get_value` are to be treated as Regexps, and implicitly add regexp anchors and case-insensitivity (i.e., a `get_value` of `'router bgp 100'` becomes the regexp `/^router bgp 100$/i`). If you want to explicitly specify a regexp (perhaps because the default behavior does not meet your needs for a specific property), be sure to add the `^` and `$` anchors to ensure you match the correct feature information in the `show` output and do not unexpectedly match similar but undesired CLI strings.
```

Comment:
If the expected model is to use the new implicit-regex style, should some (perhaps most) of the examples in the documentation be updated to use it?